src/transformers/trainer.py

```diff
@@ -2305,7 +2305,7 @@ def _load_rng_state(self, checkpoint):
                 )
                 return
 
-        checkpoint_rng_state = torch.load(rng_file)
+        checkpoint_rng_state = torch.load(rng_file, weights_only=True)
```

Comment:
I looks like this breaks loading rng_state from a saved checkpoint: https://app.circleci.com/pipelines/github/huggingface/transformers/77407/workflows/3502e84f-41fb-4f01-8a7a-2c40b9e76aa7/jobs/985682/steps